Sources/ArgumentParser/Parsing/ArgumentSet.swift

```diff
@@ -356,27 +356,35 @@ extension ArgumentSet {
     
     var result = ParsedValues(elements: [], originalInput: all.originalInput)
     var positionalValues: [(InputOrigin.Element, String)] = []
+    var unusedOptions: [(InputOrigin.Element, String)] = []
     var usedOrigins = InputOrigin()
     
     try setInitialValues(into: &result)
     
     // Loop over all arguments:
-    while let (origin, next) = set.popNext() {
+    while let (origin, next) = inputArguments.popNext() {
       defer {
-        set.removeAll(in: usedOrigins)
+        inputArguments.removeAll(in: usedOrigins)
       }
       
       switch next {
       case let .value(v):
         positionalValues.append((origin, v))
       case let .option(parsed):
+        // Look for an argument that matches this `--option` or `-o`-style
+        // input. If we can't find one, just move on to the next input. We
+        // defer catching leftover arguments until we've fully extracted all
+        // the information for the selected command.
         guard
           let argument: ArgumentDefinition = try? first(matching: parsed, at: origin)
-          else { continue }
+          else {
+            unusedOptions.append((origin, String(describing: parsed)))
```

Comment:
I’m not sure if `String(describing: parsed)` is the right approach. For #4 I’ve been adding something like
```swift
extension SplitArguments {
    func original(at origin: InputOrigin.Element) -> String {
        switch origin {
        case .argumentIndex(let index):
            return originalInput[index.inputIndex.rawValue]
        }
    }
}
```
and then use it here like
```swift
unusedOptions.append((origin, all.original(at: origin)))
```
such that this will actually get the string from the original input — regardless of how it was parsed. But I could also try to clean up these uses as part of #4 if that makes more sense.

Reply:
Way better not to set up this implicit dependency on `Parsed`'s string representation, thanks.